powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/SignatureServiceBehavior.java

```diff
@@ -335,11 +336,18 @@ private ValidateSignatureResponse validateSignature(ActivationRecordEntity activ
         List<SecretKey> signatureKeys = powerAuthServerKeyFactory.keysForSignatureType(powerAuthSignatureTypes, masterSecretKey);
 
         // Get activation version and validate it
-        Integer version = activation.getVersion();
-        if (version == null || version < 2 || version > 3) {
+        Integer signatureVersion = activation.getVersion();
+        if (signatureVersion == null || signatureVersion < 2 || signatureVersion > 3) {
             throw localizationProvider.buildExceptionForCode(ServiceError.ACTIVATION_INCORRECT_STATE);
         }
 
+        // Handle migration from version 2 to version 3, the version is forced during migration commit
```

Comment:
dtto - this feels a bit too hardcoded... If there is some logic that we need to migrate from V2 to V3, we should probably keep it isolated.